src/shared/text_or_image.rs

```diff
@@ -34,46 +36,80 @@ live_design! {
         }
         image_view = <View> {
             visible: false,
-            cursor: Default, // Use `Hand` once we support clicking on the image
+            cursor: Hand,
             width: Fill, height: Fit,
             image = <Image> {
                 width: Fill, height: Fit,
-                fit: Smallest,
+                fit: Size, // Only for a comfortable test, would set back to `Smallest` if this pr OK.
             }
         }
     }
 }
 
+#[derive(Debug, Clone, DefaultNone)]
+pub enum TextOrImageAction {
+    Click(OwnedMxcUri),
+    None,
+}
+
+#[derive(Clone)]
+pub struct ImageValue {
+    pub original_mxc_uri: OwnedMxcUri,
+    pub timeline_image_data: Arc<[u8]>,
+}
 
 /// A view that holds an image or text content, and can switch between the two.
 ///
-/// This is useful for displaying alternate text when an image is not (yet) available
+/// This is useful for displaying alternate text when an image is not (y680-121-263et) available
 /// or fails to load. It can also be used to display a loading message while an image
 /// is being fetched.
 #[derive(Live, Widget, LiveHook)]
 pub struct TextOrImage {
     #[deref] view: View,
     #[rust] status: TextOrImageStatus,
-    // #[rust(TextOrImageStatus::Text)] status: TextOrImageStatus,
     #[rust] size_in_pixels: (usize, usize),
+    #[rust] image_value: Option<ImageValue>,
 }
 
 impl Widget for TextOrImage {
     fn handle_event(&mut self, cx: &mut Cx, event: &Event, scope: &mut Scope) {
+        // We handle hit events if the status is `Image`.
+        if let TextOrImageStatus::Image = self.status {
+            let image_area = self.view.image(id!(image_view.image)).area();
+            match event.hits(cx, image_area) {
+                Hit::FingerDown(_) => {
+                    cx.set_key_focus(image_area);
+                }
+                Hit::FingerUp(fe) => {
+                    if fe.was_tap() {
```

Comment:
this needs to have more conditions on it (I realize this is confusing):
```suggestion
                Hit::FingerUp(fe) if fe.is_over && fe.is_primary_hit() && fe.was_tap() => {
```